pkg/health/server.go

```diff
@@ -33,16 +34,53 @@ type Server interface {
 }
 
 type server struct {
-	ready *atomic.Bool
-	log   logger.Logger
+	ready  *atomic.Bool
+	router http.Handler
+	log    logger.Logger
+}
+
+type RouterOptions func() (string, http.Handler)
+
+func NewRouterOptions(path string, handler http.Handler) RouterOptions {
+	return func() (string, http.Handler) {
+		return path, handler
+	}
+}
+
+func NewJsonDataRouterOptions[T any](path string, getter func() (T, error)) RouterOptions {
+	return NewRouterOptions(path, http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		data, err := getter()
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			writer.Write([]byte(err.Error()))
+			return
+		}
+		bytes, err := json.Marshal(data)
```

Comment:
Let's use streams please to avoid buffering (large) data in memory.

```go
writer.Header().Set("Content-Type", "application/json")
err := json.NewEncoder(writer).Encode(data)
// Handle error
```

Reply:
Thanks, Done.